android-gif-drawable/src/main/java/pl/droidsonroids/gif/GifDrawable.java

```diff
@@ -54,973 +50,979 @@
+     * has never been needed yet. Returned value does not change during runtime.
+     *
+     * @return possible size of the memory needed to store pixels of this object
+     */
+    public long getAllocationByteCount() {
+        long byteCount = mNativeInfoHandle.getAllocationByteCount();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            byteCount += mBuffer.getAllocationByteCount();
+        } else {
+            byteCount += getFrameByteCount();
+        }
+        return byteCount;
+    }
+
+    /**
+     * Returns the maximum possible size of the allocated memory used to store pixels and metadata of this object.
+     * It counts length of all frame buffers. Returned value does not change over time.
+     *
+     * @return maximum possible size of the allocated memory needed to store metadata of this object
+     */
+    public long getMetadataAllocationByteCount() {
+        return mNativeInfoHandle.getMetadataByteCount();
+    }
+
+    /**
+     * Returns length of the input source obtained at the opening time or -1 if
+     * length cannot be determined. Returned value does not change during runtime.
+     * If GifDrawable is constructed from {@link InputStream} -1 is always returned.
+     * In case of byte array and {@link ByteBuffer} length is always known.
+     * In other cases length -1 can be returned if length cannot be determined.
+     *
+     * @return number of bytes backed by input source or -1 if it is unknown
+     */
+    public long getInputSourceByteCount() {
+        return mNativeInfoHandle.getSourceLength();
+    }
+
+    /**
+     * Returns in pixels[] a copy of the data in the current frame. Each value is a packed int representing a {@link Color}.
+     *
+     * @param pixels the array to receive the frame's colors
+     * @throws ArrayIndexOutOfBoundsException if the pixels array is too small to receive required number of pixels
+     */
+    public void getPixels(@NonNull int[] pixels) {
+        mBuffer.getPixels(pixels, 0, mNativeInfoHandle.getWidth(), 0, 0, mNativeInfoHandle.getWidth(), mNativeInfoHandle.getHeight());
+    }
+
+    /**
+     * Returns the {@link Color} at the specified location. Throws an exception
+     * if x or y are out of bounds (negative or &gt;= to the width or height
+     * respectively). The returned color is a non-premultiplied ARGB value.
+     *
+     * @param x The x coordinate (0...width-1) of the pixel to return
+     * @param y The y coordinate (0...height-1) of the pixel to return
+     * @return The argb {@link Color} at the specified coordinate
+     * @throws IllegalArgumentException if x, y exceed the drawable's bounds
+     * @throws IllegalStateException    if drawable is recycled
+     */
+    public int getPixel(@IntRange(from = 0) int x, @IntRange(from = 0) int y) {
+        if (x >= mNativeInfoHandle.getWidth()) {
+            //need to check explicitly because reused bitmap may be larger
```

Comment:
This comment should be in L759.
After the change it seems to refer to the throw statement, not a check.
Applies also to the similar changes.